esm_parser/esm_parser.py

```diff
@@ -436,7 +458,7 @@ def attach_single_config(config, path, attach_value):
     #config.update(attachable_config)
 
 
-def attach_to_config_and_remove(config, attach_key):
+def attach_to_config_and_remove(config, attach_key, all_config=None):
```

Comment:
Is this change a best-practice thing? cause in principle the `all_config` is set to be `None` by default. Just a question, no action is required.

Reply:
This is a good point. It looks like it is a good practice in Python. Here is a good read: 
https://docs.python-guide.org/writing/gotchas/

It is always possible to check `if all_config is (or not) None:` before getting the information from `all_config`. 

The main idea is, I added some extra information in the main config and they need to be seen (readonly) in the parser. Unfortunately, in the current version they are not. So, this was a light-weight trick to pass the config. Please note that the first `config` in `attach_to_config_and_remove(config, attach_key, all_config=None)` is not always the main (big) `config`, but rather something like `config[model]` which does not contain the `config['general']` information.

Reply:
Yes, I understood that `config` could be for example `all_config[model]`, and you needed the full config. 

Also, thanks for the link you sent, it actually clarified to me some weird behavior of one of the methods I was coding.